### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by stefanomerotta** [genaray/Arch#4] [Q&A] [answered]
Hi!
I found your project while searching for a simple and fast ECS library for a new project: it seems very fast and simple to use!

Because the project is just born, do you have any roadmap for new features that will be added in the future?
Comparing to other frameworks the systems creation and orchestration are currently missing.

Do you plan to leave the ownership of creating and managing systems to the end user or will you plan to implement something to handle them?

Also do you have any plan to include queries for "just added", "just modified" and "just removed" components?

Again, compliments for this project that is very fast and easy to use, and a good codebase to learning something new from latest .NET implementation!

**Comment by genaray** [comment, marked as answer]
> Hi! I found your project while searching for a simple and fast ECS library for a new project: it seems very fast and simple to use!
> 
> Because the project is just born, do you have any roadmap for new features that will be added in the future? Comparing to other frameworks the systems creation and orchestration are currently missing.
> 
> Do you plan to leave the ownership of creating and managing systems to the end user or will you plan to implement something to handle them?
> 
> Also do you have any plan to include queries for "just added", "just modified" and "just removed" components?
> 
> Again, compliments for this project that is very fast and easy to use, and a good codebase to learning something new from latest .NET implementation!

Thanks a lot ! :) Actually makes me pretty happy that someone found this. It even got faster with the new `Highperformance Queries` since it inlines the entity logic which makes it almost 3-4 times as fast as previous.

Theres a lot planned, i probably really should add a roadmap. Currently im working on a pooled "JobScheduler" for multithreading queries to make iteration even faster. Other features in my pipeline are currently : 

- Structural changes
- Component/Entity Events
- CommandBuffer as an addition for multithreading
- Entity Templates

This repo however will focus on an efficient and easy to use archetype based ecs. "Systems" are currently basically "Queries". Since most classes are partial, i could imagine to add real systems via some sort of "add on" and another repo in the future ^^ Those could probably look like unity systems. I also accept contributions if someone is willing to take over this part :D
